common/djangoapps/third_party_auth/saml.py

```diff
@@ -25,9 +26,6 @@ def setting(self, name, default=None):
         if not hasattr(self, '_config'):
             from .models import SAMLConfiguration
             self._config = SAMLConfiguration.current()  # pylint: disable=attribute-defined-outside-init
-        if not self._config.enabled:
-            from django.core.exceptions import ImproperlyConfigured
-            raise ImproperlyConfigured("SAML Authentication is not enabled.")
```

Comment:
I think this check is a useful safety measure and worth keeping here. The main problem is that by raising a django exception instead of a python-social-auth exception, we cause a 500 instead of a 3xx or 4xx error.

What about
```
        if not self._config.enabled:
            raise SAMLBackendDisabled()
```

where that exception is defined earlier in this file as:
```python
from social.exceptions import SocialAuthBaseException
class SAMLBackendDisabled(SocialAuthBaseException):
    def __str__(self):
        return 'The SAML third party authentication backend is not enabled.'
```